enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/RequireUpperBoundDeps.java

```diff
@@ -242,18 +262,33 @@ private String getFullArtifactName( DependencyNode node, boolean usePremanaged )
         implements DependencyNodeVisitor
     {
 
-        private boolean uniqueVersions;
+        private final boolean uniqueVersions;
+        private final Set<String> ignoredDependencyScopes;
+        private final Log log;
 
-        public void setUniqueVersions( boolean uniqueVersions )
+        public RequireUpperBoundDepsVisitor( boolean uniqueVersions, List<String> ignoredDependencyScopes, Log log )
         {
             this.uniqueVersions = uniqueVersions;
+            this.ignoredDependencyScopes = ignoredDependencyScopes != null 
+                    ? new HashSet<String>( ignoredDependencyScopes ) : null;
+            this.log = log;
         }
 
         private Map<String, List<DependencyNodeHopCountPair>> keyToPairsMap =
             new LinkedHashMap<String, List<DependencyNodeHopCountPair>>();
 
         public boolean visit( DependencyNode node )
         {
+            Artifact artifact = node.getArtifact();
+            String artifactScope = artifact.getScope();
+            if ( ignoredDependencyScopes != null && ignoredDependencyScopes.contains( artifactScope ) ) 
```

Comment:
I think this filtering should rather be done in `buildErrorMessages`. It is unclear to me what the effect will be of filtering during visits to (potentially transitive) dependencies but it will probably be wrong, since the scope declared at top level is not necessarily the same as that used throughout the dependency chain.